File: BotManager/BotLib/Sources/com/rkarp/botlib/plugin/BotPlugin.java
package com.rkarp.botlib.plugin;

import com.rkarp.botlib.runner.BotRunner;

/**
 * Interface for the plugin classes.
 */
public interface BotPlugin {

	/**
	 * Performs the plugin layout node action.
	 */
	void setLayout(BotRunner runner);

	/**
	 * Performs the plugin action.
	 */
	void run(BotRunner runner);
}

File: projects/JavaFXApps/BotManager/BotLib/src/main/java/net/rickiekarp/botlib/plugin/BotPlugin.kt
package net.rickiekarp.botlib.plugin

import net.rickiekarp.botlib.runner.BotRunner

/**
 * Interface for the plugin classes.
 */
interface BotPlugin {

    /**
     * Performs the plugin layout node action.
     */
    fun setLayout(runner: BotRunner<*, *>)

    /**
     * Performs the plugin action.
     */
    fun run(runner: BotRunner<*, *>)
}
